express-notes-api/read.js

```diff
@@ -0,0 +1,16 @@
+import read from './read-file.js';
+import handleError from './handle-error.js';
+
+export default async function readNotes(id) {
+  const { notes } = await read();
+  const notesArray = Object.values(notes);
+  if (!id) {
+    return [200, notesArray];
+  } else if (isNaN(Math.floor(id)) || Math.floor(id) !== Number(id) || id <= 0) {
+    return handleError(400, 'id must be a positive integer');
```

Comment:
While this doesn't directly change the output, it's a good idea to perform the cheaper validation work (making sure the id is a positive integer) before performing more expensive work (reading from the file system), so that you can avoid the more expensive work if the input is invalid to begin with.